### PR TITLE
Document Circular Dependency + Some cleanup (dependency still here)

### PR DESCRIPTION
**Pull request by leogrim**
@ryanpbrewster @andrewconner

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/emails/EmailTemplateSender.scala [hunk 1]
```diff
@@ -10,10 +10,31 @@ import com.keepit.heimdal.{ HeimdalServiceClient, UserEventTypes, UserEvent, Hei
...
+com.keepit.common.social.SocialGraphActor (Inject) ->
+com.keepit.common.social.UserConnectionCreator (Inject) ->
+com.keepit.commanders.FriendConnectionNotifier (Inject) ->
+com.keepit.commanders.emails.FriendConnectionMadeEmailSender (Inject) ->
+com.keepit.commanders.emails.EmailTemplateSender
+
+$$$ ProdSocialGraphModule is the default implementation of SocialGraphModule defined in ShoeboxModule.   $$$
+$$$ In tests, this circular dependency can be broken at this step by using FakeSocialGraphModule instead. $$$
+*/
+
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/emails/EmailTemplateSender.scala:37]: @ryanpbrewster @andrewconner

File: kifi-backend/modules/shoebox/test/com/keepit/social/SocialUserRawInfoTest.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+class SocialUserRawInfoTest extends Specification {
+
+  "SocialUserRawInfo" should {
+    "do a basic serialization flow" in {
+
+      val info = SocialUserRawInfo(Some(Id(1)), Some(Id(2)), SocialId("leo_grimaldi"), SocialNetworks.TWITTER, "Léo Grimaldi", Stream(Json.obj("nevermind" -> "whatever")))
+      val json = SocialUserRawInfo.format.writes(info)
+      val newInfo = SocialUserRawInfo.format.reads(json).get
+      info === newInfo
+    }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/test/com/keepit/social/SocialUserRawInfoTest.scala:16]: Can't hurt.

File: kifi-backend/modules/common/app/com/keepit/social/RemoteSecureSocialUserService.scala [hunk 3]
```diff
@@ -70,13 +70,6 @@ class RemoteSecureSocialUserPlugin @Inject() (
         throw ex
     }
 
-  private var maybeSocialGraphPlugin: Option[SocialGraphPlugin] = None
-
-  @Inject(optional = true)
-  def setSocialGraphPlugin(sgp: SocialGraphPlugin) {
-    maybeSocialGraphPlugin = Some(sgp)
-  }
-
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/social/RemoteSecureSocialUserService.scala:79]: #weird #unused

File: kifi-backend/modules/shoebox/app/com/keepit/social/SocialUserRawInfo.scala [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        userId = (obj \ "userId").asOpt[Id[User]],
+        socialUserInfoId = (obj \ "socialUserInfoId").asOpt[Id[SocialUserInfo]],
+        socialId = (obj \ "socialId").as[SocialId],
+        networkType = (obj \ "networkType").as[SocialNetworkType],
+        fullName = (obj \ "fullName").as[String],
+        jsons = (obj \ "jsons").asOpt[Seq[JsValue]].map(_.toStream) getOrElse Stream()
+      )
+    }
+  }
+}
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/social/SocialUserRawInfo.scala:43]: This stuff and related classes moves out of common, to Shoebox.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/emails/EmailTemplateSender.scala (hunk 1 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/emails/EmailTemplateSender.scala:37]: :+1: 

Good to have that documented.

Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/emails/EmailTemplateSender.scala:37]: circular dependency is usually caused by several big classes that do (inject) too many things. And it gets worse when more classes begin to inject `Provider[Something]`.

**Pull request merged**